lib/markdown-preview-view.js

```diff
@@ -0,0 +1,532 @@
+const path = require('path')
+
+const { Emitter, Disposable, CompositeDisposable, File } = require('atom')
+const _ = require('underscore-plus')
+const fs = require('fs-plus')
+
+const renderer = require('./renderer')
+
+module.exports = class MarkdownPreviewView {
+  static deserialize (params) {
+    return new MarkdownPreviewView(params)
+  }
+
+  constructor ({ editorId, filePath }) {
+    this.editorId = editorId
+    this.filePath = filePath
+    this.element = document.createElement('div')
+    this.element.classList.add('markdown-preview')
+    this.element.tabIndex = -1
+    this.emitter = new Emitter()
+    this.loaded = false
+    this.disposables = new CompositeDisposable()
+    this.registerScrollCommands()
+    if (this.editorId != null) {
+      this.resolveEditor(this.editorId)
+    } else if (atom.packages.hasActivatedInitialPackages()) {
+      this.subscribeToFilePath(this.filePath)
+    } else {
+      this.disposables.add(
+        atom.packages.onDidActivateInitialPackages(() => {
+          return this.subscribeToFilePath(this.filePath)
+        })
+      )
+    }
+  }
+
+  serialize () {
+    return {
+      deserializer: 'MarkdownPreviewView',
+      filePath: this.getPath() != null ? this.getPath() : this.filePath,
+      editorId: this.editorId
+    }
+  }
+
+  copy () {
+    return new MarkdownPreviewView({
+      editorId: this.editorId,
+      filePath: this.getPath() != null ? this.getPath() : this.filePath
+    })
+  }
+
+  destroy () {
+    this.disposables.dispose()
+    return this.element.remove()
+  }
+
+  registerScrollCommands () {
+    this.disposables.add(
+      atom.commands.add(this.element, {
+        'core:move-up': () => {
+          this.element.scrollTop -= document.body.offsetHeight / 20
+        },
+        'core:move-down': () => {
+          this.element.scrollTop += document.body.offsetHeight / 20
+        },
+        'core:page-up': () => {
+          this.element.scrollTop -= this.element.offsetHeight
+        },
+        'core:page-down': () => {
+          this.element.scrollTop += this.element.offsetHeight
+        },
+        'core:move-to-top': () => {
+          this.element.scrollTop = 0
+        },
+        'core:move-to-bottom': () => {
+          this.element.scrollTop = this.element.scrollHeight
+        }
+      })
+    )
+  }
+
+  onDidChangeTitle (callback) {
+    return this.emitter.on('did-change-title', callback)
+  }
+
+  onDidChangeModified (callback) {
+    // No op to suppress deprecation warning
+    return new Disposable()
+  }
+
+  onDidChangeMarkdown (callback) {
+    return this.emitter.on('did-change-markdown', callback)
+  }
+
+  subscribeToFilePath (filePath) {
+    this.file = new File(filePath)
+    this.emitter.emit('did-change-title')
+    this.disposables.add(
+      this.file.onDidRename(() => this.emitter.emit('did-change-title'))
+    )
+    this.handleEvents()
+    return this.renderMarkdown()
+  }
+
+  resolveEditor (editorId) {
```

Comment:
Not positive on this one...but it looks like its return statements can all be removed as the caller doesn't use it